### PR TITLE
There is minimumTouchLengthToChangePage property which not used correctly

### PR DESCRIPTION
**Pull request by moadib**
There is minimumTouchLengthToChangePage property which not used correctly. Now scrolllayer use it and works as expected.

Also changed behavior when user trying to slide over first or last page. There is no need to show him empty screen. I've added marginOffset property which determine maximum margin offset which user can achieve when slide pages.

Please check code, i'm just convert my contribution to cocos2d-x version of this class.

### CODE REVIEW

File: Extensions/CCScrollLayer/CCScrollLayer.m [hunk 1]
```diff
@@ -379,33 +382,37 @@ enum
...
-	if ( (newX - startSwipe_) < -self.minimumTouchLengthToChangePage && (currentScreen_+1) < [layers_ count] )
-	{		
-		[self moveToPage: [self pageNumberForPosition:self.position] ];		
-	}
-	else if ( (newX - startSwipe_) > self.minimumTouchLengthToChangePage && currentScreen_ > 0 )
-	{		
-		[self moveToPage: [self pageNumberForPosition:self.position] ];		
+	int selectedPage = currentScreen_;
+	CGFloat delta = touchPoint.x - startSwipe_;
+	if (fabs(delta) >= m_fMinimumTouchLengthToChangePage)
```
Comment (psineur) [inline Extensions/CCScrollLayer/CCScrollLayer.m]: This line doesn't compile.

**Comment by psineur** [comment]
Hello! Thank you for sharing your work.
I tried to use your code, but looks like it's not ready yet.
There was a little compiler error, that i mentioned, but it's not a problem.

As i can see you want to create a stoping-sliding effect like in UIKit elements.
But if you started scrolling not from last/first page - this effect will not work.
I tried to use curPage dynamicaly - like in Mac version, but it only added laggish effect.

Anyway, i would like to merge such functionality, but previously it should be:
1) Compatible with old version and disabled by default.
2) It should work both on Mac/iOS.

**Comment by moadib** [comment]
You are absolutely right about incorrect behavior when scrolling started not from last/first page. 
Thank you, i fixed it in c++ version too.

Sorry for compiler error, i haven't mac, so i write this changes without possibility to test :) So i ask to help me with this.

File: Extensions/CCScrollLayer/CCScrollLayer.m [hunk 2]
```diff
@@ -379,33 +382,41 @@ enum
 	}
 	
 	if (state_ == kCCScrollLayerStateSliding)
-		self.position = ccp( (- currentScreen_ * (self.contentSize.width - self.pagesWidthOffset)) + (touchPoint.x-startSwipe_),0);	
-	
+	{
+		CGFloat desiredX = (- currentScreen_ * (self.contentSize.width - self.pagesWidthOffset)) + offset;
```
Comment (psineur) [inline Extensions/CCScrollLayer/CCScrollLayer.m]: Undeclared identifier - offset. What must be here? CGFloat offset declared first time few lines under.

**Comment by moadib** [comment]
Argh. Take a look, fixed this error.

**Comment by psineur** [comment]
Thanks, merged!

**Pull request closed without merging**